cmd/posix_test.go

```diff
@@ -35,7 +36,13 @@ func newPosixTestSetup() (StorageAPI, string, error) {
 	if err != nil {
 		return nil, "", err
 	}
-	posixStorage, err := newPosix(diskPath)
+	var endpoints []*url.URL
+	// Initialize a new posix layer.
+	endpoints, err = parseStorageEndpoints([]string{diskPath})
+	if err != nil {
+		return nil, "", err
+	}
+	posixStorage, err := newPosix(endpoints[0])
```

Comment:
if you make newPosix() accept a path instead of *url.URL above code can be avoided, also passing url.URL is not adding value to newPosix.

The same pattern can be seen everywhere else, i.e url.URL is constructed just before calling newPosix() and inside newPosix() only Path gets used